### PR TITLE
macos: build holochain for macos on demand

### PR DESCRIPTION
**Pull request by DavHau**
### Summary
Like https://github.com/holochain/holochain/pull/1760 but trimmed down to whats absolutely necessary for achieving a reproducible build for macos

### TODO:
- [ ] CHANGELOG(s) updated with appropriate info
- [ ] Just before pressing the merge button, ensure new entries to CHANGELOG(s) are still under the _UNRELEASED_ heading

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -0,0 +1,45 @@
+name: build holochain
+on:
+  workflow_dispatch:
+  pull_request:
```
Comment (steveeJ) [inline .github/workflows/build.yml:4]:
```suggestion
```

**Review by steveeJ — COMMENTED** [review]
this is great stuff!

let's disable this on PRs for now so they don't occupy the arm64 runner which is still a scarce resource :smile:

File: .github/workflows/build.yml (hunk 1 above)
Comment (DavHau) [reply .github/workflows/build.yml]: BTW, it requires only 2 minutes even if it has to re-build holochain.

Comment (steveeJ) [reply .github/workflows/build.yml]: ah i see,that's probably when it doesn't have to rebuild the dependencies, right?

**Review by steveeJ — APPROVED** [review]
:partying_face:

**Pull request merged**